src/Makefile.am

```diff
@@ -359,6 +370,51 @@ libbitcoin_wallet_tool_a_SOURCES = \
   wallet/wallettool.cpp \
   $(BITCOIN_CORE_H)
 
+# rustc only reqires the main .rs file as input, but rustc must also be invoked
+# if any of the other files change. List the main file first, then targets can
+# depend on $(LIBBITCOIN_RUSTY_SRCS), but only build the first source file ($<).
+LIBBITCOIN_RUSTY_SRCS= \
+  rusty/src/lib.rs \
+  rusty/src/bridge.rs \
+  rusty/src/dns_headers.rs \
+  rusty/src/rest_downloader.rs \
+  rusty/src/p2p_addrs.rs rusty/src/p2p_client.rs \
+  rusty/src/p2p_socket_handler.rs
+
+LIBBITCOIN_RUSTY_TEST_SRCS= \
+  rusty/src/test_bridge.rs
+
+rusty/liblibc.rlib: rusty/libc/src/lib.rs $(wildcard rusty/libc/src/*.rs)
+	$(RUSTC) --crate-name libc $< --crate-type=rlib -g -O -C overflow-checks=on -C panic=unwind --out-dir $(@D) && \
+	touch $@
+
+rusty/libbech32.rlib: rusty/rust-bech32/src/lib.rs $(wildcard rusty/rust-bech32/src/*.rs)
+	$(RUSTC) --crate-name bech32 $< --crate-type=rlib -g -O -C overflow-checks=on -C panic=unwind --out-dir $(@D) && \
+	touch $@
+
+rusty/libbitcoin_hashes.rlib: rusty/bitcoin_hashes/src/lib.rs $(wildcard rusty/bitcoin_hashes/src/*.rs)
+	$(RUSTC) --crate-name bitcoin_hashes $< --crate-type=rlib -g -O -C overflow-checks=on -C panic=unwind --cfg 'feature="default"' --cfg 'feature="std"' --out-dir $(@D) && \
+	touch $@
+
+rusty/libsecp256k1.rlib: rusty/rust-secp256k1/src/lib.rs $(wildcard rusty/rust-secp256k1/src/*.rs) $(LIBSECP256K1)
+	$(RUSTC) --crate-name secp256k1 $< --crate-type=rlib -g -O -C overflow-checks=on -C panic=unwind --cfg 'feature="std"' --cfg 'feature="dont_replace_c_symbols"' --out-dir $(@D) && \
+	touch $@
+
+rusty/libbitcoin.rlib: rusty/libbech32.rlib rusty/libbitcoin_hashes.rlib rusty/libsecp256k1.rlib
+rusty/libbitcoin.rlib: rusty/rust-bitcoin/src/lib.rs $(wildcard rusty/rust-bitcoin/src/*.rs) $(wildcard rusty/rust-bitcoin/src/*/*.rs) $(wildcard rusty/rust-bitcoin/src/*/*/*.rs)
+	$(RUSTC) --crate-name bitcoin $< --crate-type=rlib -g -O -C overflow-checks=on -C panic=unwind --out-dir $(@D) --extern bech32=$(@D)/libbech32.rlib --extern bitcoin_hashes=$(@D)/libbitcoin_hashes.rlib --extern secp256k1=$(@D)/libsecp256k1.rlib && \
+	touch $@
+
+# We force overflow checks on so that any integer overflows result in a panic
+# instead of, eg, attempting to alloc 4GB of memory (though we will catch such
+# an alloc and refuse it, generating a panic instead).
+# Also force panics to unwind (instead of calling abort()) as we want to catch
+# panics in the Rust threads and simply close the thread, instead of bringing
+# down the rest of Bitcoin Core.
+$(LIBBITCOIN_RUSTY): $(LIBBITCOIN_RUSTY_SRCS) rusty/libbitcoin.rlib rusty/liblibc.rlib
+	$(RUSTC) --crate-name rusty $< --target=$(RUST_TARGET) --crate-type=staticlib --emit=link -g -O -C lto=on -C overflow-checks=on -C panic=unwind --out-dir $(@D) -L dependency=$(@D) --extern libc=$(@D)/liblibc.rlib --extern bitcoin_hashes=$(@D)/libbitcoin_hashes.rlib --extern bitcoin=$(@D)/libbitcoin.rlib
```

Comment:
`-L rusty` pointing to directory should resolve rlibs automatically instead of using all these explicit `--extern`s?

Reply:
Doesn't seem to for me, no?

Reply:
I tested and it seems like `--extern` is only necessary for `libc` (`rustc` hits [E0658](https://doc.rust-lang.org/error-index.html#E0658) otherwise). This works for me:
`/home/igor/.cargo/bin/rustc --crate-name rusty rusty/src/lib.rs --target=x86_64-unknown-linux-gnu --crate-type=staticlib --emit=link -g -C lto=on -C overflow-checks=on -C panic=unwind --out-dir rusty -L rusty --extern libc=rusty/liblibc.rlib`

Reply:
Ehh, ight as well just leave it, then.

Reply:
Yeah, it's not a big deal